modules/external-secrets/data.tf

```diff
@@ -14,15 +11,18 @@ data "aws_iam_policy_document" "external_secrets" {
       "secretsmanager:DescribeSecret",
       "secretsmanager:ListSecretVersionIds",
     ]
-    resources = concat(
-      var.external_secrets_secrets_manager_arns,
-      ["arn:${var.addon_context.aws_partition_id}:secretsmanager:${var.addon_context.aws_region_name}:${var.addon_context.aws_caller_identity_account_id}:secret:*"]
```

Comment:
This was previously hard-coding `:secret:*` that could not be configured using `var.external_secrets_secrets_manager_arns`, permitting wide-open access